### ISSUE TITLE
Dynamo crashes at LoopWhile

### ISSUE THREAD

**Issue by MarcinTata** [DynamoDS/Dynamo#9095]
## Dynamo version

2.0.1.5055

## Operating system

W10

## What did you do? 

Run the script with 2 packages:
- Structural Analysis 
- Optimo

## What did you expect to see?

An result from a loopWhile

## What did you see instead?

Can anyone tell me where is the issue? What went wrong? Shall I keep posting errors like this one?

"Określone rzutowanie jest nieprawidłowe. PL "
"The specified cast is incorrect." eng 
   w ProtoCore.DSASM.Heap.ToHeapObject[T](StackValue heapObject)
   w ProtoCore.DSASM.Heap.RecursiveMark(StackValue root)
   w ProtoCore.DSASM.Heap.SingleStep(Boolean forceGC)
   w ProtoCore.DSASM.Executive.POP_helper(Instruction instruction, Int32& blockId, Int32& dimensions)
   w ProtoCore.DSASM.Executive.POP_Handler(Instruction instruction)
   w ProtoCore.DSASM.Executive.Execute(Int32 exeblock, Int32 entry, Language language)
   w ProtoCore.DSASM.Executive.Execute(Int32 exeblock, Int32 entry, List`1 breakpoints, Language language)
   w ProtoCore.DSASM.Interpreter.Run(Int32 codeblock, Int32 entry, Language lang, List`1 breakpoints)
   w ProtoCore.Lang.JILFunctionEndPoint.Execute(Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.CallSite.ExecWithZeroRI(List`1 functionEndPoint, Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore, SingleRunTraceData previousTraceData, SingleRunTraceData newTraceData)
   w ProtoCore.CallSite.Execute(List`1 functionEndPoint, Context c, List`1 formalParameters, List`1 replicationInstructions, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.CallSite.DispatchNew(Context context, List`1 arguments, List`1 partialReplicationGuides, DominantListStructure domintListStructure, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.Lang.FunctionPointerEvaluator.Evaluate(List`1 args, StackFrame stackFrame)
   w ProtoCore.Lang.ArrayUtilsForBuiltIns.Evaluate(StackValue function, StackValue parameters, StackValue unpackParams, Interpreter runtime, StackFrame stackFrame)
   w ProtoCore.Lang.BuiltInFunctionEndPoint.Execute(Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.CallSite.ExecWithZeroRI(List`1 functionEndPoint, Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore, SingleRunTraceData previousTraceData, SingleRunTraceData newTraceData)
   w ProtoCore.CallSite.Execute(List`1 functionEndPoint, Context c, List`1 formalParameters, List`1 replicationInstructions, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.CallSite.DispatchNew(Context context, List`1 arguments, List`1 partialReplicationGuides, DominantListStructure domintListStructure, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.DSASM.Executive.Callr(Int32 blockDeclId, Int32 functionIndex, Int32 classIndex, Boolean& explicitCall, Boolean isDynamicCall, Boolean hasDebugInfo)
   w ProtoCore.DSASM.Executive.CALLR_Handler(Instruction instruction)
   w ProtoCore.DSASM.Executive.Execute(Int32 exeblock, Int32 entry, Language language)
   w ProtoCore.DSASM.Executive.Execute(Int32 exeblock, Int32 entry, List`1 breakpoints, Language language)
   w ProtoCore.DSASM.Interpreter.Run(Int32 codeblock, Int32 entry, Language lang, List`1 breakpoints)
   w ProtoCore.Lang.JILFunctionEndPoint.Execute(Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.CallSite.ExecWithZeroRI(List`1 functionEndPoint, Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore, SingleRunTraceData previousTraceData, SingleRunTraceData newTraceData)
   w ProtoCore.CallSite.Execute(List`1 functionEndPoint, Context c, List`1 formalParameters, List`1 replicationInstructions, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.CallSite.DispatchNew(Context context, List`1 arguments, List`1 partialReplicationGuides, DominantListStructure domintListStructure, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.Lang.FunctionPointerEvaluator.Evaluate(List`1 args, StackFrame stackFrame)
   w ProtoCore.Lang.ArrayUtilsForBuiltIns.Evaluate(StackValue function, StackValue parameters, StackValue unpackParams, Interpreter runtime, StackFrame stackFrame)
   w ProtoCore.Lang.BuiltInFunctionEndPoint.Execute(Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.CallSite.ExecWithZeroRI(List`1 functionEndPoint, Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore, SingleRunTraceData previousTraceData, SingleRunTraceData newTraceData)
   w ProtoCore.CallSite.Execute(List`1 functionEndPoint, Context c, List`1 formalParameters, List`1 replicationInstructions, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.CallSite.DispatchNew(Context context, List`1 arguments, List`1 partialReplicationGuides, DominantListStructure domintListStructure, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.DSASM.Executive.Callr(Int32 blockDeclId, Int32 functionIndex, Int32 classIndex, Boolean& explicitCall, Boolean isDynamicCall, Boolean hasDebugInfo)
   w ProtoCore.DSASM.Executive.CALLR_Handler(Instruction instruction)
   w ProtoCore.DSASM.Executive.Execute(Int32 exeblock, Int32 entry, Language language)
   w ProtoCore.DSASM.Executive.Execute(Int32 exeblock, Int32 entry, List`1 breakpoints, Language language)
   w ProtoCore.DSASM.Interpreter.Run(Int32 codeblock, Int32 entry, Language lang, List`1 breakpoints)
   w ProtoCore.Lang.JILFunctionEndPoint.Execute(Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.CallSite.ExecWithZeroRI(List`1 functionEndPoint, Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore, SingleRunTraceData previousTraceData, SingleRunTraceData newTraceData)
   w ProtoCore.CallSite.ExecWithRISlowPath(List`1 functionEndPoint, Context c, List`1 formalParameters, List`1 replicationInstructions, StackFrame stackFrame, RuntimeCore runtimeCore, SingleRunTraceData previousTraceData, SingleRunTraceData newTraceData)
   w ProtoCore.CallSite.ExecWithRISlowPath(List`1 functionEndPoint, Context c, List`1 formalParameters, List`1 replicationInstructions, StackFrame stackFrame, RuntimeCore runtimeCore, SingleRunTraceData previousTraceData, SingleRunTraceData newTraceData)
   w ProtoCore.CallSite.Execute(List`1 functionEndPoint, Context c, List`1 formalParameters, List`1 replicationInstructions, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.CallSite.DispatchNew(Context context, List`1 arguments, List`1 partialReplicationGuides, DominantListStructure domintListStructure, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.DSASM.Executive.Callr(Int32 blockDeclId, Int32 functionIndex, Int32 classIndex, Boolean& explicitCall, Boolean isDynamicCall, Boolean hasDebugInfo)
   w ProtoCore.DSASM.Executive.CALLR_Handler(Instruction instruction)
   w ProtoCore.DSASM.Executive.Execute(Int32 exeblock, Int32 entry, Language language)
   w ProtoCore.DSASM.Executive.Execute(Int32 exeblock, Int32 entry, List`1 breakpoints, Language language)
   w ProtoCore.DSASM.Interpreter.Run(Int32 codeblock, Int32 entry, Language lang, List`1 breakpoints)
   w ProtoCore.Lang.JILFunctionEndPoint.Execute(Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.CallSite.ExecWithZeroRI(List`1 functionEndPoint, Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore, SingleRunTraceData previousTraceData, SingleRunTraceData newTraceData)
   w ProtoCore.CallSite.Execute(List`1 functionEndPoint, Context c, List`1 formalParameters, List`1 replicationInstructions, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.CallSite.DispatchNew(Context context, List`1 arguments, List`1 partialReplicationGuides, DominantListStructure domintListStructure, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.Lang.FunctionPointerEvaluator.Evaluate(List`1 args, StackFrame stackFrame)
   w ProtoCore.Lang.ArrayUtilsForBuiltIns.Evaluate(StackValue function, StackValue parameters, StackValue unpackParams, Interpreter runtime, StackFrame stackFrame)
   w ProtoCore.Lang.BuiltInFunctionEndPoint.Execute(Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.CallSite.ExecWithZeroRI(List`1 functionEndPoint, Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore, SingleRunTraceData previousTraceData, SingleRunTraceData newTraceData)
   w ProtoCore.CallSite.Execute(List`1 functionEndPoint, Context c, List`1 formalParameters, List`1 replicationInstructions, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.CallSite.DispatchNew(Context context, List`1 arguments, List`1 partialReplicationGuides, DominantListStructure domintListStructure, StackFrame stackFrame, RuntimeCore runtimeCore)
   w ProtoCore.DSASM.Executive.Callr(Int32 blockDeclId, Int32 functionIndex, Int32 classIndex, Boolean& explicitCall, Boolean isDynamicCall, Boolean hasDebugInfo)
   w ProtoCore.DSASM.Executive.CALLR_Handler(Instruction instruction)
   w ProtoCore.DSASM.Executive.Execute(Int32 exeblock, Int32 entry, Language language)
   w ProtoCore.DSASM.Executive.Execute(Int32 exeblock, Int32 entry, List`1 breakpoints, Language language)
   w ProtoCore.DSASM.Executive.BounceUsingExecutive(Executive executive, Int32 exeblock, Int32 entry, StackFrame stackFrame, Int32 locals, Boolean fepRun, Executive exec, List`1 breakpoints)
   w ProtoScript.Runners.ProtoScriptRunner.ExecuteLive(Core core, RuntimeCore runtimeCore)
   w ProtoScript.Runners.LiveRunner.Execute(Boolean isCodeCompiled)
   w ProtoScript.Runners.LiveRunner.CompileAndExecute(List`1 astList)
   w ProtoScript.Runners.LiveRunner.CompileAndExecuteForDeltaExecution(List`1 astList)
   w ProtoScript.Runners.LiveRunner.SynchronizeInternal(GraphSyncData syncData)
   w ProtoScript.Runners.LiveRunner.UpdateGraph(GraphSyncData syncData)
   w Dynamo.Scheduler.UpdateGraphAsyncTask.HandleTaskExecutionCore()
   w Dynamo.Scheduler.AsyncTask.Execute()

**Comment by aparajit-pratap** [comment]
@MarcinTata could you attach the script or describe the steps to reproduce this crash with the above packages installed.

**Comment by MarcinTata** [comment]
The script has some custom nodes and require another programm: Autodesk Roboto Stuructural Analysis.
How Optimo works is presented here https://www.youtube.com/watch?v=oc7znmsJ_vw 
You set an amount of "population"(e.g 10) and number of iterations (runs of loopWhile)  and then the loop begins, generating new scores. So 10 x 4 should produce 40 scores.
Basically my scipt look the same, except fitness function.
The problem seems to be in RAM memory requirements. Dynamo stops after generating 23 scores. For example  when I set 5x5, Dynamo stops at 5th iteration, when I set 25x4, Dynamo stops at 1st.

NOTE, not every time I get popout with error, like I posted before. Sometimes it just stop working like everything was calculated properly, but it isn't, there's still some iterations left.

I can send you the script if it is necessary.
